packages/manager/src/routes/routes.test.tsx

```diff
@@ -1,54 +1,70 @@
-import {
-  RouterProvider,
-  createMemoryHistory,
-  createRouter,
-} from '@tanstack/react-router';
-import { waitFor } from '@testing-library/react';
+import { RouterProvider } from '@tanstack/react-router';
+import { screen, waitFor } from '@testing-library/react';
 import React from 'react';
 
-import { accountSettingsFactory } from 'src/factories';
 import { renderWithTheme } from 'src/utilities/testHelpers';
 
-import { routeTree } from './index';
-import { allPaths } from './utils/allPaths';
-
-import type { RouterContext } from './types';
-
-// Mock any context or dependencies your routes might need
-const mockContext: RouterContext = {
-  accountSettings: accountSettingsFactory.build(),
-  isACLPEnabled: false,
-  isDatabasesEnabled: false,
-  isPlacementGroupsEnabled: false,
-};
-
-// Helper function to create a test router
-const createTestRouter = (initialPath: string) => {
-  return createRouter({
-    context: mockContext,
-    history: createMemoryHistory({ initialEntries: [initialPath] }),
-    routeTree,
-  });
-};
+import { migrationRouter } from './index';
+import { getAllRoutePaths } from './utils/allPaths';
+
+const allMigrationPaths = getAllRoutePaths(migrationRouter);
+
+vi.mock('src/hooks/useFlags', () => ({
+  useFlags: () => ({
+    selfServeBetas: true,
+  }),
+}));
 
-describe.skip('Route Tests', () => {
-  test.each(allPaths)('renders %s route correctly', async (path) => {
-    const router = createTestRouter(path);
+describe('Migration Router', () => {
+  const renderWithRouter = (initialEntry: string) => {
+    migrationRouter.invalidate();
+    migrationRouter.navigate({ replace: true, to: initialEntry });
 
-    const { findByRole, getByRole } = renderWithTheme(
-      <RouterProvider router={router} />
+    return renderWithTheme(
+      <React.Suspense>
+        <RouterProvider router={migrationRouter} />
+      </React.Suspense>
     );
```

Comment:
Can we use the `flags` option rather than using `vi.mock('src/hooks/useFlags')`?

```suggestion
    return renderWithTheme(
      <React.Suspense>
        <RouterProvider router={migrationRouter} />
      </React.Suspense>,
      { flags: { selfServeBetas: true } }
    );
```